mobile/src/screens/ProfileScreen.tsx

```diff
@@ -13,18 +14,86 @@ import { I18nContext } from '../context/I18nContext';
 import { Language, Languages } from '../models/Languages';
 import { AuthContext } from '../context/AuthContext';
 import { APIContext } from '../context/APIContext';
+import { Book } from '../models/Book';
+import { PaginatedBookList } from '../components/PaginatedBookList';
+import { useIsFocused } from '@react-navigation/native';
 
+const BOOKS_PER_PAGE = 9; 
+// or to simulate column book list, BOOKS_PER_PAGE = favorites.length + (3 - favorites.length % 3)
+
+/**
+ * Saved Tab to display user's favorited books
+ */
 const SavedTab: React.FC = () => {
+
+  const client = useContext(APIContext);
+  const auth = useContext(AuthContext);
+  const i18n = useContext(I18nContext);
+
+  // Favorite/saved books state
+  const [favorites, setFavorites] = useState<Book[]>([]);
+
+  // determines if the saved tab is unfocused/focused
+  const isFocused = useIsFocused();
+
+  // Rerender tab and get updated favorites when isFocused changes
+  // (Ensure that Profile AND Book Screens have synchronised favorites info)
+  useEffect(
+    () => {
+      (async () => {
+        try {
+          if (auth.user !== null) {
+            const res = await client.getFavorites();
+            setFavorites(res);
+          }
+        } catch (e) {
+          console.log(e.message);
+        }
+      })();
+    }, [isFocused]
+  );
```

Comment:
Let's cache this, for offline support and to fix the flashing "Tap :bookmark: to save a book" message when you first load the page.